gbpservice/nfp/orchestrator/drivers/orchestration_driver.py

```diff
@@ -751,14 +820,10 @@ def get_network_function_device_status(self, device_data,
             raise exceptions.ComputePolicyNotSupported(
                 compute_policy=device_data['service_details']['device_type'])
 
-        token = self._get_token(device_data.get('token'))
-        if not token:
-            return None
-
         try:
             device = self.compute_handler_nova.get_instance(
-                            token,
-                            self._get_admin_tenant_id(token=token),
+                            device_data['token'],
+                            device_data['tenant_id'],
```

Comment:
Check, the earlier code in the same function was doing something like this,         
token = self._get_token(device_data.get('token'))
        if not token:
            return None
So, we are already expecting device_data to have token, we had it but we never passed it and were fetching again, costly, as it involves REST APIs.